modules/@apostrophecms/piece-type/index.js

```diff
@@ -295,8 +270,6 @@ module.exports = {
           }
           return self.publish(req, draft);
         },
-        // TEMP - This works fine, but should be reviewed during work actually
-        // focused on batch archive/restore.
         async archive (req) {
```

Comment:
This was flagged temp and now it's not, so I'm calling it here, although I should apologize for not spotting it earlier when it was first PR'd in:

The iteration function here is going straight to the database, which is not right because the afterArchive event will not fire. Also, permissions are skipped.

The correct approach is to find the document via the module, set the archived property of the document ot true, and then call `update` through Apostrophe's model layer. The model layer will automatically figure out if it's newly archived and emit appropriate events.

Reply:
🤦 duh